happi/client.py

```diff
@@ -600,16 +603,24 @@ def _store(self, device, insert=False):
         post.update({'type': tpe,
                      'creation': creation,
                      'last_edit': ttime.ctime()})
+
         # Find id
         try:
-            _id = post[self._id_key]
+            # In case we want to update the name of an entry
+            # We want to add a new entry, and delete the old one
+            if the_old_name != post[self._id_key]:
+                _id = the_old_name
+                self.backend.save(post[self._id_key], post, insert=True)
+                self.backend.delete(the_old_name)
+            else:
+                _id = post[self._id_key]
+                self.backend.save(_id, post, insert=insert)
         except KeyError:
             raise EntryError('HappiItem did not supply the proper information '
                              'to interface with the database, missing {}'
                              ''.format(self._id_key))
         # Store information
         logger.info('Adding / Modifying information for %s ...', _id)
```

Comment:
The logger message should be moved up to the individual `save` and `delete` calls, indicating what's happening. e.g.,
`Saving new entry...`
`Removing old entry...`

Reply:
Fixed this in b69e45d